prime-router/metadata/fhir_mapping/hl7/datatype/HumanName.yml

```diff
@@ -46,11 +46,19 @@ suffix_3:
   generateList: true
   valueOf: XCN.21
   expressionType: HL7Spec
-     
+
+
 period:
-    valueOf: datatype/Period
+  valueOf: datatype/Period
+  expressionType: resource
+  vars:
+    start: XPN.12 | XCN.19
+    end: XPN.13 | XCN.20
+
+period_2:
+    valueOf: datatypes/DR/Period
+    condition: XPN.12 NULL && XPN.13 NULL
     expressionType: resource
     vars:
-       start: XPN.12 | XCN.19
-       end: XPN.13 | XCN.20
-     
\ No newline at end of file
+       start: XPN.10.1
+       end: XPN.10.2
```

Comment:
We had a slack thread going on whether to use a separate file for period or use nested elements. In my PR I went with the nested element strategy like this:
```
period:
  generateList: true
  expressionType: nested
  vars:
    start: XTN.13
    end: XTN.14
  expressionsMap:
    start:
      condition: $start NOT_NULL
      type: STRING
      valueOf: "GeneralUtils.dateTimeWithZoneId(start,ZONEID)"
      expressionType: JEXL
    end:
      condition: $end NOT_NULL
      type: STRING
      valueOf: "GeneralUtils.dateTimeWithZoneId(end,ZONEID)"
      expressionType: JEXL
```

I don't have a strong opinion on which strategy to use but I do think we should norm on it. What do you think?

Reply:
Even if it is used in multiple places as it will eventually be? I thought the point was to have reusable code so it wouldn't need to be maintained in multiple places

Reply:
Yeah that was crux of the thread: https://agilesix.slack.com/archives/C026DV8R21W/p1697730231591709 (you'll probs have to change the url to nava.slack.com). You're definitely right that period is used all over the place so it could go in a datatypes/common folder. Plus there are probably other elements similar to this that could be added to common datatypes. But Kalish made an argument that the nested style keeps the mapping more aligned with the mapping inventory since the inventory doesn't reference a specific Period mapping. I leaned that way myself but it's just a preference thing.